### PR TITLE
[chore] enable renovatebot for some deps updates

### PR DESCRIPTION
**Pull request by codeboten**
This moves updates for docker/docker-compose/github actions away from dependabot and over to renovatebot. Renovatebot will group PRs for those dependencies in a single pull request per dependency group.

Related issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19410

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1102,3 +1092,13 @@ updates:
     schedule:
       interval: "weekly"
       day: "wednesday"
+  - package-ecosystem: "gomod"
+    directory: "/receiver/zipkinreceiver"
```
Comment (TylerHelmuth) [inline .github/dependabot.yml:1096]: Are these being added because they were missing and shouldve been here the whole time?

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      {
+        "matchManagers": ["github-actions"],
+        "groupName": "github-actions deps"
+      },
+      {
+        "matchManagers": ["gomod"],
+        "enabled": false
+      }
+    ]
+  }
```
Comment (TylerHelmuth) [inline renovate.json:28]:
```suggestion
  }

```

**Review by TylerHelmuth — APPROVED** [review]

**Comment by codeboten** [comment]
Confirmed that the output from gendependabot used to say
```
The following modules are not included in the dependabot file because it has a limit of 220 entries:
  - /receiver/zipkinreceiver
  - /receiver/zookeeperreceiver
  - /testbed
  - /testbed/mockdatareceivers/mockawsxrayreceiver
  - /testbed/mockdatasenders/mockdatadogagentexporter
```

and now says

```
The following modules are not included in the dependabot file because it has a limit of 220 entries:
  - /testbed
  - /testbed/mockdatareceivers/mockawsxrayreceiver
  - /testbed/mockdatasenders/mockdatadogagentexporter
```

File: .github/dependabot.yml (hunk 1 above)
Comment (codeboten) [reply .github/dependabot.yml:1096]: Correct... these are added because two items are removed from the top

**Review by mx-psi — APPROVED** [review]

**Comment by trask** [comment]
@codeboten I'm very interested in your experience in this repo with renovate, and I suspect other maintainers would be also, could you share in the maintainers meeting after it has been running in this repo for a couple of weeks?

**Pull request merged**